src/main/java/net/snowflake/ingest/streaming/internal/BlobBuilder.java

```diff
@@ -89,13 +90,21 @@ static <T> Blob constructBlobAndMetadata(
         final byte[] compressedChunkData;
         final int chunkLength;
         final int compressedChunkDataSize;
+        int extendedMetadataSize = -1;
 
         if (internalParameterProvider.getEnableChunkEncryption()) {
           Pair<byte[], Integer> paddedChunk =
               padChunk(serializedChunk.chunkData, Constants.ENCRYPTION_ALGORITHM_BLOCK_SIZE_BYTES);
           byte[] paddedChunkData = paddedChunk.getFirst();
           chunkLength = paddedChunk.getSecond();
 
+          if (internalParameterProvider.getComputeExtendedMetadataSize()) {
+            extendedMetadataSize =
+                Utils.getLittleEndianInt(
```

Comment:
Use the BytesUtils methods from the parquet library instead?

Reply:
Changed to `ByteUtils` instead of own func.